File: self-dividing-numbers/self-dividing-numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int>res ;
        for(int i = left ; i <= right ; i++){
            int num = i ;
            int  ans = 0 ;
            while(num!=0){
                int rem = num%10;
                if(rem == 0 || i % rem != 0){
                    ans = 1 ;
                    break ; 
                }
                num = num / 10 ;
            }
            if(ans==0){
            res.push_back(i);
        }
        }
        return res;
    }
};

File: self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for i in range(left,right+1):
            num = i
            ans = True
            if "0" not in str(i):
                while num:
                    rem = num%10
                    if i%rem != 0:
                        ans = False
                    num = num//10
                if ans:
                    res.append(i)
        return res

